firebase-inappmessaging/src/androidTest/java/com/google/firebase/inappmessaging/FirebaseInAppMessagingFlowableTest.java

```diff
@@ -223,22 +225,31 @@ public void setUp() {
     clearProtoDiskCache(InstrumentationRegistry.getTargetContext());
     application =
         spy((Application) InstrumentationRegistry.getTargetContext().getApplicationContext());
-    String id = FirebaseInstanceId.getInstance().getId();
-    when(instanceId.getId()).thenReturn(id);
-    when(instanceId.getInstanceId())
+    when(firebaseInstallations.getId()).thenReturn(Tasks.forResult(INSTANCE_ID));
+    when(firebaseInstallations.getToken(false))
         .thenReturn(
             Tasks.forResult(
-                new InstanceIdResult() {
+                new InstallationTokenResult() {
```

Comment:
Why anonymous subclass? How about `InstallationTokenResult.builder()...build()`?

Reply:
Brilliant, I didn't notice this was a builder!